src/Silk.Data/MediatR/Handlers/TagRequestHandler.cs

```diff
@@ -0,0 +1,166 @@
+            public DeleteHandler(SilkDbContext db)
+            {
+                _db = db;
+            }
+            
+            public async Task<Unit> Handle(TagRequest.Delete request, CancellationToken cancellationToken)
+            {
+                Tag tag = await _db
+                    .Tags
+                    .Include(t => t.OriginalTag)
+                    .Include(t => t.Aliases)
+                    .AsSplitQuery()
+                    .FirstOrDefaultAsync(t =>
+                        t.Name.ToLower() == request.Name.ToLower() &&
+                        t.GuildId == request.GuildId, cancellationToken);
+                
+                Tag[] aliasedTags = await _db
+                    .Tags
+                    .Include(t => t.OriginalTag)
+                    .Include(t => t.Aliases)
+                    .AsSplitQuery()
+                    .Where(t => t.OriginalTagId == tag.Id || t.OriginalTag!.OriginalTagId == tag.Id)
+                    .ToArrayAsync(cancellationToken);
+                
+                
+                foreach (Tag t in aliasedTags)
+                    _db.Tags.Remove(t);
+                
+                _db.Tags.Remove(tag);
+                await _db.SaveChangesAsync(cancellationToken);
+                return new();
+            }
+        }
+        
+        public class GetByUserHandler : IRequestHandler<TagRequest.GetByUser, IEnumerable<Tag>?>
+        {
+            private readonly SilkDbContext _db;
+            public GetByUserHandler(SilkDbContext db)
+            {
+                _db = db;
+            }
+            
+            public async Task<IEnumerable<Tag>?> Handle(TagRequest.GetByUser request, CancellationToken cancellationToken)
+            { 
+                Tag[] tags = await _db
+                    .Tags
+                    .Include(t => t.OriginalTag)
+                    .Include(t => t.OriginalTag)
+                    .Where(t => t.GuildId == request.GuildId && t.OwnerId == request.OwnerId)
+                    .ToArrayAsync(cancellationToken);
+                
+                return tags.Any() ? tags : null; // Return null over empty list //
+            }
+        }
+
+        public class GetByNameHandler : IRequestHandler<TagRequest.GetByName, IEnumerable<Tag>?>
+        {
+            private readonly SilkDbContext _db;
+            public GetByNameHandler(SilkDbContext db)
+            {
+                _db = db;
+            }
+
+            public async Task<IEnumerable<Tag>?> Handle(TagRequest.GetByName request, CancellationToken cancellationToken)
+            {
+                Tag[] tags = await
+                    _db
+                        .Tags
+                        .Include(t => t.Aliases)
+                        .Include(t => t.OriginalTag)
+                        .AsSplitQuery()
+                        .Where(t => EF.Functions.Like(t.Name.ToLower(), request.Name.ToLower() + '%'))
```

Comment:
I'm pretty sure `Like` is case insensitive by default.  I suppose it depends on the actual DB used.

Reply:
Postgres is not, by default unfortunately.